modules/virt-adding-vm-to-service-mesh.adoc

```diff
@@ -0,0 +1,100 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-connecting-vm-to-service-mesh.adoc
+
+[id="virt-adding-vm-to-service-mesh_{context}"]
+= Configuring a virtual machine for the service mesh
+
+To add a virtual machine (VM) workload to a service mesh, enable automatic sidecar injection in the VM configuration file by setting the `sidecar.istio.io/inject` annotation to `true`. Then expose your VM as a service to view your application in the mesh.
+
+
+.Procedure
+
+. Edit the VM configuration file to add the `sidecar.istio.io/inject: "true"` annotation.
++
+.Example configuration file
+[source,yaml]
+----
+  apiVersion: kubevirt.io/v1
+  kind: VirtualMachine
+  metadata:
+    labels:
+      kubevirt.io/vm: vm-istio
+    name: vm-istio
+  spec:
+    runStrategy: Always
+    template:
+      metadata:
+        labels:
+          kubevirt.io/vm: vm-istio
+          app: vm-istio <1>
+        annotations:
+          sidecar.istio.io/inject: "true" <2>
+      spec:
+        domain:
+          devices:
+            interfaces:
+            - name: default
+              masquerade: {} <3>
+            disks:
+            - disk:
+                bus: virtio
+              name: containerdisk
+            - disk:
+                bus: virtio
+              name: cloudinitdisk
+          resources:
+            requests:
+              memory: 1024M
+        networks:
+        - name: default
+          pod: {}
+        terminationGracePeriodSeconds: 0
+        volumes:
+        - containerDisk:
+            image: registry:5000/kubevirt/fedora-cloud-container-disk-demo:devel
+          name: containerdisk
+----
+<1> The key/value pair (label) that must be matched to the service selector attribute.
+<2> The annotation to enable automatic sidecar injection.
+<3> The binding method (masquerade mode) for use with the default pod network.
++
+[NOTE]
+====
+To avoid port conflicts with sidecars, ports used by Istio proxy should not be used by user workloads.
+====
+
+. Apply the VM configuration:
++
+[source,terminal]
+----
+$ oc apply -f <vm_name>.yaml <1>
+----
+<1> The name of the virtual machine YAML file.
+
+
+. Create a `Service` object to expose your VM to the service mesh.
++
+[source,yaml]
+----
+  apiVersion: v1
+  kind: Service
+  metadata:
+    name: vm-istio
+  spec:
+    selector:
+      app: vm-istio <1>
+    ports:
+      - port: 8080
+        name: http
+        protocol: TCP
+----
+<1> The service selector which determines the set of pods targeted by a service. This attribute corresponds to the `spec.metadata.labels` field in the VM configuration file. In the above example the `Service` object named `vm-istio` targets TCP port 8080 on any pod with the label `app=vm-istio`.
```

Comment:
* which -> that
* Suggest a comma after "In the above example"

Reply:
Fixed